### DISCUSSION TITLE
It got worse after adapt

### DISCUSSION THREAD

**Discussion by MrLuer** [ChristianOrr/Real-time-self-adaptive-deep-stereo#2] [General]
Thank you for your great work, I used kitti pre-trained weights and it got worse after adapt on kitti left and right images

before
![24model7_colorized](https://github.com/ChristianOrr/madnet-deep-stereo-with-keras/assets/33979932/e6278635-ff17-43cc-8658-e4adf387c5c7)

after
![24model7_colorized](https://github.com/ChristianOrr/madnet-deep-stereo-with-keras/assets/33979932/6afc8e85-23a0-443d-b7c8-1a6c03d79a6e)

**Comment by ChristianOrr** [comment]
Hi Mr Leur,

I don't know what you did, but I can confirm there are situations where it can perform worse with MAD adaptation. If you overtrain for example the model can start to degrade. If there is a large domain shift it can also perform worse, like for example with a new stereo system. 

MAD performs well when it's used to do minor alterations to the model's weights. You can think of MAD as being very overfitting sensitive.

**Comment by MrLuer** [comment]
Thank you for your previous reply. Now I have another question. It seems that mask is not taken into account in the code?

**Comment by ChristianOrr** [comment]
Can you elaborate?

**Reply by MrLuer** [reply]
Stereo data consider occlusion issues, loss and EPE calculations will exclude occlusions through masks.
